src/anndata/_core/sparse_dataset.py

```diff
@@ -639,6 +639,10 @@ class _CSCDataset(BaseCompressedSparseDataset, abc.CSCDataset):
     """Internal concrete version of :class:`anndata.abc.CSRDataset`."""
 
 
+abc.CSRDataset.register(_CSRDataset)
+abc.CSCDataset.register(_CSCDataset)
```

Comment:
This is a deep cut for me - don't these both already inherit from the ABC? I'd like to understand this one.  Does it have to do with the multiple inheritance?

```python
from anndata._core.sparse_dataset import _CSRDataset
from anndata.abc import CSRDataset
assert issubclass(_CSRDataset, CSRDataset)
```

Reply:
huh, you’re right.

I must have run into something else then, no idea why I thought that check wasn’t already right.